client/lib/warn/index.js

```diff
@@ -1,14 +1,13 @@
 /**
  * Internal Dependencies
  */
-var config = require( 'config' );
+import config from 'config';
 
-function warn() {
-	if ( config( 'env' ) !== 'production' ) {
```

Comment:
>We can use also `process.env.NODE_ENV !== 'production'`

It's my understanding that we're not to use `NODE_ENV` and instead rely on the value specified in `config`, which is equal to the current environment's `CALYPSO_ENV` environment variable [specified in the Dockerfile](https://github.com/Automattic/wp-calypso/blob/bb1b6c32a4d3e6237fbe0f56bf2d541922728853/Dockerfile#L32-L35) for each environment's build.

>the block should be removed during compilation.

I'm not sure the uglifier is that smart. A common approach to achieve this is to use something like [Webpack's `DefinePlugin`](https://webpack.github.io/docs/list-of-plugins.html#defineplugin) to replace occurrences of a text with `false`, which the uglifier would be smart enough to remove.

```js
// 1. Variable comparison
var env = 'production';
if ( env !== 'production' ) {
	// Could be, but probably not removed by uglifier
}

// 2. Use `DefinePlugin` to replace `__DEV__` with `false`
if ( false ) {
	// Definitely removed by uglifier
}
```